include/npcomp/Dialect/Torch/IR/OpInterfaces.td

```diff
@@ -0,0 +1,42 @@
+//===- OpInterface.td --------------------------------------*- tablegen -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+include "mlir/IR/OpBase.td"
+
+#ifndef NPCOMP_DIALECT_ATEN_IR_TORCH_OP_INTERFACES
+#define NPCOMP_DIALECT_ATEN_IR_TORCH_OP_INTERFACES
+
+def TorchKernelOpInterface : OpInterface<"TorchKernelOpInterface"> {
+  let description = [{
+    This interface provides metadata about ops that are isomorphic with a
+    Torch kernel call. It is implemented by both the generic torch.kernel_call
+    op and any recognized ops (i.e. in the ATen and other dialects) which
+    are transformed from it and remain isomorphic with a Torch kernel.
+  }];
+
+  let methods = [
+    InterfaceMethod<"Metadata for the kernel",
+      "::mlir::NPCOMP::Torch::KernelMetadata", "getTorchKernelMetadata">,
+  ];
+}
+
+def TorchBuildableKernelOpInterface :
+    OpInterface<"TorchBuildableKernelOpInterface"> {
+  let description = [{
+    Implemented by specific ops that can be constructed from a generic
```

Comment:
Is it "Provides the metadata to drive the conversion." or "Provides *additional* metadata on top of TorchKernelOpInterface ..."? The comments on BuildKernelMetadata suggest the latter.

Reply:
Updated comments a lot. Let me know if it makes any more sense.

Reply:
Looks a lot better, thanks!